src/test/java/io/github/christophermanahan/carnitas/HTTPServerTest.java

```diff
@@ -130,36 +138,90 @@ public TestConnection accept() {
             this.closed = false;
         }
 
-        public Optional<String> receive() {
-            return closed ? Optional.empty() : Optional.of(request);
+        public Receiver receiver() {
+            return closed ? new TestReceiver(null) : new TestReceiver(request);
         }
 
-        public void send(Response httpResponse) {
-            response = new String(httpResponse.serialize());
+        public void send(Response response) {
+            this.response = new String(response.serialize());
         }
 
         public void close() {
             closed = true;
         }
     }
 
-    private class SendException extends TestConnection {
-        SendException(String request) {
-            super(request);
+    private class TestReceiver implements Receiver {
```

Comment:
This feels like overkill. I think it is a symptom of the `Connection` depending on the a `Receiver` and exposing it in the API. What can we do about that?

Reply:
Change the API to be consistent with a connection abstraction!